Batches to Intacct plugin (#134)
KingdomFirst/RockBlocks: Intacct/ReadMe.md

![Kingdom First Solutions](../.screenshots/KFSBanner.jpg)

# Intacct Export to Journal 
_Tested/Supported in Rock Version:  13.0-14.0_    
_Released:  10/9/2018_   
_Updated:  4/13/2023_   

## Summary 

This plug in will allow you to create journal entries in Intacct for batches with the click of a button.

Quick Links:
- [What's New](#whats-new)
- [Configuration](#configuration)
    - [Intacct Configuration](#intacct-configuration)
    - [Rock Configuration](#rock-configuration)
- [Advanced Configuration](#advanced-configuration)

## What's New

The following new goodness will be added to your Rock install with this plugin:

> - **New Page**: Intacct Export (Finance > Functions > Intacct Export)  
> - **New Block**: Batch to Journal (added to the Batch Detail Page on install)  
> - **New Block**: Batches to Journal (added to the new Intacct Export Page on install)  
> - **New Account Attributes**: There are a number of new Account attributes that control where transactions are posted in Intacct  
> - **New Page**: Intacct Projects (Finance > Administration > Intacct Projects)  

Reviewer: Let's try to keep like items together, so moving Intacct Projects New Page up to the top.
```suggestion
- **New Page**: Intacct Projects (Finance > Administration > Intacct Projects)  
- **New Page**: Intacct Export (Finance > Functions > Intacct Export)  
- **New Block**: Batch to Journal (added to the Batch Detail Page on install)  
- **New Block**: Batches to Journal (added to the new Intacct Export Page on install)  
- **New Account Attributes**: There are a number of new Account attributes that control where transactions are posted in Intacct  
```

>     Log Response: Flag indicting if the Intacct Response should be logged to the Batch Audit Log

Reviewer: ```suggestion
    Log Response: Flag indicating if the Intacct Response should be logged to the Batch Audit Log
```

    
    User Password: The Intacct API password. This is the same information you use when you log into the Sage UI.
    
    Location Id: The optional Intacct Location Id. Add a location ID to log into a multi-entity shared company. Entities are typically different locations of a single company.
    
    Export Mode: Determines the type of object to create in Intacct. Selecting Journal Entry will result in creating journal entries of the type set in the Journal Id setting. Selecting Other Receipts will result in creating Other Receipts in the Cash Management area of Intacct.
```
<div style="page-break-after: always;"></div>

**Batches to Journal Block**

The Batches to Journal block was added to a new page under your Finances heading named Intacct Export. It consists of a specialized grid of exportable batches with an export button at the bottom.

![](../.screenshots/BatchToIntacct/BatchesToJournalBlock.png)

When the block is in Other Receipts mode, you will have options for Deposit To, Payment Method and Bank Account next to the Export to Intacct button.

![](../.screenshots/BatchToIntacct/BatchesToJournalBlockOtherReceipts.png)

> You will need to configure the Batch to Journal block settings.

Reviewer: ```suggestion
You will need to configure the Batches to Journal block settings.
```

![](../.screenshots/BatchToIntacct/BatchesToJournalBlockSettings.png)
```
    Name: Block name

    Detail Page: The Financial Batch Detail page.

    Button Text: Customize the text for the export button.

    Months Back: Number of months back that batches should be loaded. This is helpful to prevent database timeouts if there are years of historical batches.

    Close Batch: Flag indicating if the Financial Batch should be closed in Rock when successfully posted to Intacct.
    

>     Log Response: Flag indicting if the Intacct Response should be logged to the Batch Audit Log.

Reviewer: ```suggestion
    Log Response: Flag indicating if the Intacct Response should be logged to the Batch Audit Log.
```

>     Enable Debug: Turns on/off the Lava debug panel.

Reviewer: ```suggestion
    Enable Debug: Turns on/off the Lava debug panel. The panel will show after export.
```

    
    Field Type: Text
```
<div style="page-break-after: always;"></div>

**Important Information about Your Custom Dimension**
- The Name is for internal (Rock) purposes only
- The Category has to be set to the Intacct Export category in order to be included in the API post

> - The Key is the actual Intacct specific name of the custom Dimension in all caps. Also, you'll notice that the core KFS Dimensions use the format `com.kfs.Intacct.CLASID`. The custom key can be anything you'd like, so long as there is a period before the Dimension name. For example, `org.mychurch.Intacct.MYCUSTOMDIM` is a valid Attribute Key.
> - Since the Dimensions for Accounts are typically plain text, only text attributes are supported for the custom Dimensions

Reviewer: Please add back these changes that are needed in the documentation if they didn't exist, the readme went into better detail about custom dimensions.
```suggestion
- The Key is the actual Intacct specific name of the Custom Dimension in all caps, beginning with "GLDIM". Also, you'll notice that the core KFS Dimensions use the format `rocks.kfs.Intacct.CLASSID`. The custom key can be anything you'd like, so long as there is a period before the Dimension name. For example, `org.mychurch.Intacct.GLDIMMYCUSTOMDIM` is a valid Attribute Key.
- Text attributes are recommended for the Custom Dimensions. However, the value you enter in the attribute on the Financial Account must be the System Info > ID, not the text value. (i.e. Value: MT-Event-P1234, ID: 10005. You must enter 10005 in Rock.)

**Examples:**
- *Custom Dimension 1:*
  - Record Name: Restriction
  - Integration Name: restriction
  - Rock Key: GLDIMRESTRICTION
- *Custom Dimension 2:*
  - Record Name: Custom Project
  - Integration Name: custom_project
  - Rock Key: GLDIMCUSTOM_PROJECT
- *Custom Dimension 2 Value:*
  - Custom Project: MT - Event - P1234
  - ID: 10005
  - Rock Value: 10005

![IntacctCustomDimensionSystemInfoScreenshot](https://user-images.githubusercontent.com/2990519/174348414-42ead26f-0dd5-4c3c-b985-d1fb48141508.jpeg)
```
